crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -460,15 +427,78 @@ impl<C: Catalog, A: Authorizer + Clone, S: SecretStore>
             None
         };
 
+        let storage_credentials = storage_config.as_ref().map(|c| {
+            vec![StorageCredential {
+                prefix: table_location.to_string(),
+                config: c.creds.clone().into(),
+            }]
+        });
+
         let load_table_result = LoadTableResult {
             metadata_location: metadata_location.as_ref().map(ToString::to_string),
             metadata: table_metadata,
-            config: storage_config.map(Into::into),
+            config: storage_config.map(|c| c.config.into()),
+            storage_credentials,
         };
 
         Ok(load_table_result)
     }
 
+    async fn load_table_credentials(
+        parameters: TableParameters,
+        data_access: DataAccess,
+        state: ApiContext<State<A, C, S>>,
+        request_metadata: RequestMetadata,
+    ) -> Result<LoadCredentialsResponse> {
+        // ------------------- VALIDATIONS -------------------
+        let TableParameters { prefix, table } = parameters;
+        let warehouse_id = require_warehouse_id(prefix)?;
+
+        let list_flags = ListFlags {
+            include_active: true,
+            include_staged: false,
+            include_deleted: false,
+        };
+
+        let (mut t, table_id, storage_permissions) = Self::load_prelude(
```

Comment:
I don't like the scope of the load_prelude function. Maybe we can remove transaction creation from it, add the NoStoragePermissions Error to it and give it a better name, such as `require_data_action`.
I was missing authorization here and wouldn't expect it to happen in a prelude.